DFC.ServiceTaxonomy.GraphSync/Services/SynonymService.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using DFC.ServiceTaxonomy.GraphSync.Queries;
+using DFC.ServiceTaxonomy.GraphSync.Services.Interface;
+using DFC.ServiceTaxonomy.Neo4j.Services;
+
+namespace DFC.ServiceTaxonomy.GraphSync.Services
+{
+    public class SynonymService : ISynonymService
+    {
+        private readonly IGraphDatabase _neoGraphDatabase;
+
+        public SynonymService(IGraphDatabase neoGraphDatabase)
+        {
+            _neoGraphDatabase = neoGraphDatabase ?? throw new ArgumentNullException(nameof(neoGraphDatabase));
+        }
+
+        public IEnumerable<string> GetSynonyms(string node)
+        {
+            var query = new MatchSynonymsQuery();
+
+            if(node.Equals("Skill", StringComparison.CurrentCultureIgnoreCase))
+            {
+                query = new MatchSynonymsQuery("esco__Skill", "ncs__SkillLabel", "skos__prefLabel", "ncs__hasAltLabel", "ncs__hasPrefLabel", "ncs__hasHiddenLabel");
```

Comment:
We should get these values, if we can, from the appropriate settings using `GraphSyncHelper`, rather than hard-coding them.

Reply:
Source nodes now come from Graph Sync Helper. The rest can be updated later on when GraphSyncHelper is extended